src/EditorFeatures/TestUtilities/NavigateTo/AbstractNavigateToTests.cs

```diff
@@ -70,14 +70,29 @@ protected async Task TestAsync(TestHost testHost, string content, Func<TestWorks
             await TestAsync(content, body, testHost, null);
             await TestAsync(content, body, testHost, w => new FirstDocIsVisibleDocumentTrackingService(w.Workspace));
             await TestAsync(content, body, testHost, w => new FirstDocIsActiveAndVisibleDocumentTrackingService(w.Workspace));
+
+            async Task TestAsync(
+                string content, Func<TestWorkspace, Task> body, TestHost testHost,
+                Func<HostWorkspaceServices, IDocumentTrackingService> createTrackingService)
+            {
+                using var workspace = CreateWorkspace(content, testHost, createTrackingService);
+                await body(workspace);
+            }
         }
 
-        private async Task TestAsync(
-            string content, Func<TestWorkspace, Task> body, TestHost testHost,
-            Func<HostWorkspaceServices, IDocumentTrackingService> createTrackingService)
+        protected async Task TestAsync(TestHost testHost, XElement content, Func<TestWorkspace, Task> body)
         {
-            using var workspace = CreateWorkspace(content, testHost, createTrackingService);
-            await body(workspace);
+            await TestAsync(content, body, testHost, null);
+            await TestAsync(content, body, testHost, w => new FirstDocIsVisibleDocumentTrackingService(w.Workspace));
+            await TestAsync(content, body, testHost, w => new FirstDocIsActiveAndVisibleDocumentTrackingService(w.Workspace));
+
+            async Task TestAsync(
```

Comment:
```suggestion
            await TestAsync(content, body, testHost, w => new FirstDocIsActiveAndVisibleDocumentTrackingService(w.Workspace));
            
            return;

            async Task TestAsync(
```

IDE style is to have an explicit exit point before the local funtions to make it clear that no actual method code follows htem.